pkg/sql/crdb_internal.go

```diff
@@ -2971,18 +2996,22 @@ CREATE TABLE crdb_internal.partitions (
 	subzone_id INT -- references a subzone id in the crdb_internal.zones table
 )
 	`,
-	populate: func(ctx context.Context, p *planner, dbContext *DatabaseDescriptor, addRow func(...tree.Datum) error) error {
+	generator: func(ctx context.Context, p *planner, dbContext *DatabaseDescriptor) (virtualTableGenerator, func(), error) {
 		dbName := ""
 		if dbContext != nil {
 			dbName = dbContext.Name
 		}
-		return forEachTableDescAll(ctx, p, dbContext, hideVirtual, /* virtual tables have no partitions*/
-			func(db *DatabaseDescriptor, _ string, table *TableDescriptor) error {
-				return table.ForeachNonDropIndex(func(index *sqlbase.IndexDescriptor) error {
-					return addPartitioningRows(ctx, p, dbName, table, index, &index.Partitioning,
-						tree.DNull /* parentName */, 0 /* colOffset */, addRow)
+		worker := func(addRow func(...tree.Datum) error) error {
+			return forEachTableDescAll(ctx, p, dbContext, hideVirtual, /* virtual tables have no partitions*/
+				func(db *DatabaseDescriptor, _ string, table *TableDescriptor) error {
+					return table.ForeachNonDropIndex(func(index *sqlbase.IndexDescriptor) error {
+						return addPartitioningRows(ctx, p, dbName, table, index, &index.Partitioning,
+							tree.DNull /* parentName */, 0 /* colOffset */, addRow)
+					})
 				})
-			})
+		}
+		cleanup, next := setupGenerator(worker)
```

Comment:
another nit from me: can we arrange the arguments to be the same order as the return

Reply:
done